notebooks/value_semantics/01_value_semantics_in_swift.ipynb

```diff
@@ -0,0 +1,949 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "func incrementTwoInts(_ x: inout Int, _ y: inout Int) {\n",
+    "    x += 1\n",
+    "    y += 1\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "a: 2, b: 3\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "var a = 1\n",
+    "var b = 2\n",
+    "incrementTwoInts(&a, &b)\n",
+    "print(\"a: \\(a), b: \\(b)\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "",
+     "evalue": "",
+     "output_type": "error",
+     "traceback": [
+      "error: <Cell 30>:1:22: error: inout arguments are not allowed to alias each other\nincrementTwoInts(&a, &a) // Compile error!\n                     ^~\n\n<Cell 30>:1:18: note: previous aliasing argument\nincrementTwoInts(&a, &a) // Compile error!\n                 ^~\n\n"
+     ]
+    }
+   ],
+   "source": [
+    "incrementTwoInts(&a, &a) // Compile error!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "extension Int {\n",
+    "    // Define a silly function that allows us to play around with closures.\n",
+    "    mutating func increment(by amount: (Self) -> Self) {\n",
+    "        let amt = amount(self)\n",
+    "        self += amt\n",
+    "    }\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2\r\n",
+      "4\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "// Example usage of the API defined above.\n",
+    "print(a)\n",
+    "a.increment { $0 }  // equivalent to a + a\n",
+    "print(a)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "",
+     "evalue": "",
+     "output_type": "error",
+     "traceback": [
+      "error: <Cell 33>:3:5: error: overlapping accesses to 'b', but modification requires exclusive access; consider copying to a local variable\n    b.increment { bCopy in\n    ^\n\n<Cell 33>:4:9: note: conflicting access is here\n      b = 0  // Exclusivity violation detected @ compile time!\n      ~~^~~\n\n"
+     ]
+    }
+   ],
+   "source": [
+    "func exclusivityViolationAtCompileTime() {\n",
+    "    var b = 3\n",
+    "    b.increment { bCopy in\n",
+    "      b = 0  // Exclusivity violation detected @ compile time!\n",
+    "      return bCopy\n",
+    "    }\n",
+    "}\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Simultaneous accesses to 0x7fc44a77c420, but modification requires exclusive access.\r\n",
```

Comment:
This output shows a runtime detection, which is not what the example says it illustrates.

Reply:
In the JSON format, the output comes before the cell source. Are you using the ReviewNB extension?

Reply:
The wha?